src/libraries/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -291,6 +291,34 @@ public unsafe void ToLogFont(object logFont, Graphics graphics)
             }
         }
 
+        public unsafe void ToLogFont<T>(T logFont, Graphics graphics)
+        {
+            ArgumentNullException.ThrowIfNull(logFont);
+
+            Type type = logFont.GetType();
+            int nativeSize = sizeof(Interop.User32.LOGFONT);
+            if (Marshal.SizeOf<T>() != nativeSize)
+            {
+                // If we don't actually have an object that is LOGFONT in size, trying to pass
+                // it to GDI+ is likely to cause an AV.
+                throw new ArgumentException(null, nameof(logFont));
+            }
+
+            Interop.User32.LOGFONT nativeLogFont = ToLogFontInternal(graphics);
+
+            // PtrToStructure requires that the passed in object not be a value type.
+            if (!type.IsValueType)
```

Comment:
You can remove the `type` variable and use `typeof(T).IsValueType' here.

Reply:
I do even better, and put constraints on `T` to be `struct`. I do not think it's reasonable to allow objects if you are caring about trimmability.